docs/cli/doc.md

```diff
@@ -0,0 +1,83 @@
+
+If you find any awkward rendering or unexpected behavior, refer to this document and see if a convention better addresses your specific use case.
+
+## Homepage
+
+We encourage you to customize the experience of your user documentation by writing a custom Markdown document which can be embedded at the root of your generated documentation by specifying it on the command line when generating the documentation. Every page contains links back to the homepage. If no homepage is provided, your users will be faced with an empty screen stating "There's nothing to see on this page".
+
+Currently, we do not offer a way to include arbitrary assets, so unfortunately you cannot embed a custom logo or add pages other than a homepage. We're working on this feature though, so be sure to follow along with our development!
+
+## Distributing generated documentation
+
+The generated documentation directory (named `docs` by default) is completely self contained and can be moved, zipped, and shared without any of the raw WDL files it documents. Please let us know if you run into any issues while sharing your documentation.
+
+## Custom themes
+
+While it is technically possible to supply your own custom CSS styling, this capability is currently undocumented. We recommend you stick with the default styling at this point in time, but do let us know what kinds of customization you would like to see in future releases! 
+
+## Using preamble comments for file-level documentation
+
+To provide top-level documentation for a file, add a comment block before the `version` statement where each line starts with a double pound sign (i.e., `##`, which we term a "preamble comment"). These preamble comments will be rendered as Markdown above the generated table of contents on that file's dedicated page. For example:
+
+```wdl
+## # This is a header
+##
+## This is a paragraph with **bolding**, _italics_, and `code` formatting.
+
+version 1.2
+
+workflow foo {}
+```
+
+## v1.0 and v1.1 structs
+
+The WDL specification does not offer a way to document structs prior to WDL v1.2, so the pages for them are rather limited. The pages for these structs will have a copy of the raw WDL definition.
+
+## v1.2 structs
+
+HTML documentation for structs defined in v1.2 WDL has not yet been implemented, so they are given the same treatment as pre-v1.2 structs.
+
+## Meta entries with special handling
+
+All meta entries will render in the final HTML documentation, but there are some special conventions we introduce. Each key below is expected to have a WDL `String` value.
+
+`description`
+: Every struct, task, and workflow `meta` section should have a `description` key. This description string can have Markdown formatting. The `description` string should be less than 140 characters or it will be clipped in some contexts.
```

Comment:
Note to self: ensure this `X characters` length matches the final implementation